### PR TITLE
 Remove deprecated transport settings

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #36652. There are a number of transport settings that
were deprecated and replaced with new versions in 7.x. This commit
removes them from 8.0.

### CODE REVIEW

File: docs/reference/migration/migrate_8_0/http.asciidoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[float]
+[[breaking_80_http_changes]]
+=== HTTP changes
+
+[float]
+==== Removal of old HTTP settings
+
+The `http.tcp_no_delay` setting was deprecated in 7.x and has been removed in 8.0. It has been replaced by
+`http.tcp.no_delay`.
```
Comment (lcawl) [inline docs/reference/migration/migrate_8_0/http.asciidoc]: We typically add a "deprecated[7.1.0]" to the setting's description in the release where it's deprecated, per https://github.com/elastic/docs#additions-and-deprecations
I expected to find http.tcp_no_delay defined in https://www.elastic.co/guide/en/elasticsearch/reference/7.x/modules-http.html or https://www.elastic.co/guide/en/elasticsearch/reference/7.x/modules-network.html 
@esdocs Is there somewhere else this setting is defined?

File: docs/reference/migration/migrate_8_0/network.asciidoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[float]
+[[breaking_80_network_changes]]
+=== Network changes
+
+[float]
+==== Removal of old network settings
+
+The `network.tcp.connect_timeout` setting was deprecated in 7.x and has been removed in 8.0. This setting
+was a fallback setting for `transport.connect_timeout`. To change the default connection timeout for client
+connections `transport.connect_timeout` should be modified.
```
Comment (lcawl) [inline docs/reference/migration/migrate_8_0/network.asciidoc]: Ditto re deprecation comment.  Couldn't find this setting defined either, expected it to be in https://www.elastic.co/guide/en/elasticsearch/reference/7.x/modules-network.html
If it's elsewhere we should mark it as deprecated in the 7.x branch

**Comment by lcawl** [comment]
Do you consider these changes to be notable enough to include in the Installation and Upgrade Guide too? (https://www.elastic.co/guide/en/elastic-stack/master/elasticsearch-breaking-changes.html) If so, I'll add the appropriate tags

File: docs/reference/migration/migrate_8_0/http.asciidoc (hunk 1 above)
Comment (tbrooks8) [reply docs/reference/migration/migrate_8_0/http.asciidoc]: The documentation for these settings was updated to refer to the new settings in 6.7 (including 7.0, 7.x, and 8.0). The commit to officially deprecate them was added to 7.1. What approach should I take in this scenario?

As for the http socket settings, the socket settings appear to be inconsistently documented. They are documented on the network page and for profiles on the transport page. But not on the http page or for the non-profile based transport settings. 

I guess this is a case that we need to improve?

File: docs/reference/migration/migrate_8_0/network.asciidoc (hunk 2 above)
Comment (tbrooks8) [reply docs/reference/migration/migrate_8_0/network.asciidoc]: The network connect timeout setting appears to not be documented.

**Comment by tbrooks8** [comment]
As for the upgrade guide - `transport.port`, `transport.compression`, and `transport.connect_timeout` are pretty big and maybe should be documented in a visible manner. The others are small.

File: docs/reference/migration/migrate_8_0/http.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/migration/migrate_8_0/http.asciidoc]: I had a quick chat with the @es-docs folks and they said from a support perspective, documenting that they’re no longer available makes sense, even if we never officially documented them in the first place. And it seems like including them in “breaking changes” for release X (where they were deprecated) and X+1 (where they were discontinued) is appropriate.

**Review by lcawl — APPROVED** [review]
Documentation LGTM

**Pull request merged**